tests/test_valueunit.py

```diff
@@ -0,0 +1,38 @@
+#
+
+import pytest
+import sdsc
+
+@pytest.mark.parametrize("params,result",
+  (
+    # 0 - simple, correct example
+    ((0, '100mm'),'100'),
+    # 1 - simple correct example, value explicitly expected
+    ((0, '100mm', 'value'),'100'),
+    # 2 - simple correct example, unit expected
+    ((0, '100mm', 'unit'),'mm'),
+    # 3 - malformatted example (extra space), error code expected
+    ((0, '100 mm', 'unit'),'MALFORMED_VALUE'),
+    # 4 - malformatted example (extra *), error code expected
+    ((0, '*100mm', 'value'),'MALFORMED_VALUE'),
+    # 5 - malformatted example (extra *), error code expected
+    ((0, '100mm*', 'value'),'MALFORMED_VALUE'),
+    # 6 - correct example with %, unit expected
+    ((0, '200%', 'unit'),'%'),
+    # 7 - example with bad unit %%, error code expected
+    ((0, '1%%', 'unit'),'MALFORMED_VALUE'),
+    # 8 - example with bad unit remm, error code expected
+    ((0, '77.7remm', 'unit'),'MALFORMED_VALUE'),
+    # 9 - example with float, value expected
+    ((0, '77.7rem', 'value'),'77.7'),
+    # 10 - example without unit, empty unit string expected
+    ((0, '77', 'unit'),''),
+    # 11 - example without unit, value expected
+    ((0, '91.5'),'91.5'),
+    # 12 - empty string, error code expected
+    ((0, ''),'MALFORMED_VALUE'),
+  )
+)
+def test_splitvalueunit(params,result):
+    """Checks whether values and units are properly split"""
+    assert sdsc.splitvalueunit(*params) == result
```

Comment:
Actually, I wanted to use this initially:  `    assert sdsc.splitvalueunit(0, *params) == result`
(As the `0` for the context variable is always the same.)

However, unpacking does not seem to work like this.